pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -343,3 +349,54 @@ func (s *Server) makeFileMonitor(fileDir string, domainSuffix string, configCont
 
 	return nil
 }
+
+// getTransportCredentials attempts to create credentials.TransportCredentials from ClientTLSSettings in mesh config
+// Implemented only for SIMPLE_TLS mode
+// TODO:
+//
+//	Implement CRL and SANs override for cert verification
+//	Implement for MUTUAL_TLS/ISTIO_MUTUAL_TLS modes
+func (s *Server) getTransportCredentials(args *PilotArgs, tlsSettings *v1alpha3.ClientTLSSettings) (credentials.TransportCredentials, error) {
+	switch tlsSettings.GetMode() {
+	case v1alpha3.ClientTLSSettings_SIMPLE:
+		if tlsSettings.GetInsecureSkipVerify().GetValue() {
+			return credentials.NewTLS(&tls.Config{
+				ServerName:         tlsSettings.GetSni(),
+				InsecureSkipVerify: tlsSettings.GetInsecureSkipVerify().GetValue(), //nolint
+			}), nil
+		}
+		if len(tlsSettings.GetCredentialName()) > 0 {
+			if len(tlsSettings.GetCaCertificates()) > 0 || len(tlsSettings.GetCaCrl()) > 0 {
```

Comment:
The error is about caCert and credName but we are looking at CaCrl. What is the reason for CRL?